src/futures/account.rs

```diff
@@ -0,0 +1,252 @@
+use crate::account::{OrderSide, TimeInForce};
+
+
+#[derive(Clone)]
+pub struct FuturesAccount {
+    pub client: Client,
+    pub recv_window: u64,
+}
+
+pub enum ContractType {
+    Perpetual,
+    CurrentMonth,
+    NextMonth,
+    CurrentQuarter,
+    NextQuarter,
+}
+
+impl From<ContractType> for String {
+    fn from(item: ContractType) -> Self {
+        match item {
+            ContractType::Perpetual => String::from("PERPETUAL"),
+            ContractType::CurrentMonth => String::from("CURRENT_MONTH"),
+            ContractType::NextMonth => String::from("NEXT_MONTH"),
+            ContractType::CurrentQuarter => String::from("CURRENT_QUARTER"),
+            ContractType::NextQuarter => String::from("NEXT_QUARTER"),
+        }
+    }
+}
+
+pub enum PositionSide {
+    Both,
+    Long,
+    Short,
+}
+
+impl From<PositionSide> for String {
+    fn from(item: PositionSide) -> Self {
+        match item {
+            PositionSide::Both => String::from("BOTH"),
+            PositionSide::Long => String::from("LONG"),
+            PositionSide::Short => String::from("SHORT"),
+        }
+    }
+}
+
+pub enum OrderType {
+    Limit,
+    Market,
+    Stop,
+    StopMarket,
+    TakeProfit,
+    TakeProfitMarket,
+    TrailingStopMarket,
+
+}
+
+impl From<OrderType> for String {
+    fn from(item: OrderType) -> Self {
+        match item {
+            OrderType::Limit => String::from("LIMIT"),
+            OrderType::Market => String::from("MARKET"),
+            OrderType::Stop => String::from("STOP"),
+            OrderType::StopMarket => String::from("STOP_MARKET"),
+            OrderType::TakeProfit => String::from("TAKE_PROFIT"),
+            OrderType::TakeProfitMarket => String::from("TAKE_PROFIT_MARKET"),
+            OrderType::TrailingStopMarket => String::from("TRAILING_STOP_MARKET"),
+        }
+    }
+}
+
+pub enum WorkingType {
+    MarkPrice,
+    ContractPrice,
+}
+
+impl From<WorkingType> for String {
+    fn from(item: WorkingType) -> Self {
+        match item {
+            WorkingType::MarkPrice => String::from("MARK_PRICE"),
+            WorkingType::ContractPrice => String::from("CONTRACT_PRICE"),
+        }
+    }
+}
+
+struct OrderRequest {
+    pub symbol: String,
+    pub side: OrderSide,
+    pub position_side: Option<PositionSide>,
+    pub order_type: OrderType,
+    pub time_in_force: Option<TimeInForce>,
+    pub qty: Option<f64>,
+    pub reduce_only: Option<bool>,
+    pub price: Option<f64>,
+    pub stop_price: Option<f64>,
+    pub close_position: Option<bool>,
+    pub activation_price: Option<f64>,
+    pub callback_rate: Option<f64>,
+    pub working_type: Option<WorkingType>,
+    pub price_protect: Option<f64>,
+}
+
+
+impl FuturesAccount {
+    pub fn limit_buy(&self,
+                     symbol: impl Into<String>,
+                     qty: impl Into<f64>,
+                     price: f64,
+                     time_in_force: TimeInForce,
+    ) -> Result<Order> {
```

Comment:
This will fail because the return does not have the same fields as Order.
```suggestion
    ) -> Result<Transaction> {
```
The futures api also needs it's own `Transaction` model. It is different from the spot one.
I will create an MR to this branch with my changes soon.